exp/api/v1alpha4/azuremachinepool_types.go

```diff
@@ -111,6 +129,95 @@ type (
+		// machines (ex: 10%).
+		// Absolute number is calculated from percentage by rounding down.
+		// This can not be 0 if MaxSurge is 0.
+		// Defaults to 0.
+		// Example: when this is set to 30%, the old MachineSet can be scaled
+		// down to 70% of desired machines immediately when the rolling update
+		// starts. Once new machines are ready, old MachineSet can be scaled
+		// down further, followed by scaling up the new MachineSet, ensuring
+		// that the total number of machines available at all times
+		// during the update is at least 70% of desired machines.
+		// +optional
+		// +kubebuilder:default:=0
+		MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
+
+		// The maximum number of machines that can be scheduled above the
+		// desired number of machines.
+		// Value can be an absolute number (ex: 5) or a percentage of
+		// desired machines (ex: 10%).
+		// This can not be 0 if MaxUnavailable is 0.
+		// Absolute number is calculated from percentage by rounding up.
+		// Defaults to 1.
+		// Example: when this is set to 30%, the new MachineSet can be scaled
+		// up immediately when the rolling update starts, such that the total
+		// number of old and new machines do not exceed 130% of desired
+		// machines. Once old machines have been killed, new MachineSet can
+		// be scaled up further, ensuring that total number of machines running
+		// at any time during the update is at most 130% of desired machines.
+		// +optional
+		// +kubebuilder:default:=1
+		MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty"`
+
+		// DeletePolicy defines the policy used by the MachineDeployment to identify nodes to delete when downscaling.
+		// Valid values are "Random, "Newest", "Oldest"
+		// When no value is supplied, the default is Oldest
+		// +optional
+		// +kubebuilder:validation:Enum=Random;Newest;Oldest
+		// +kubebuilder:default:=Oldest
+		DeletePolicy AzureMachinePoolDeletePolicyType `json:"deletePolicy,omitempty"`
+	}
+
+	// AzureDefaultingImage defines information about the image to use for VMSS VM creation.
+	// There are three ways to specify an image: by ID, Marketplace Image or SharedImageGallery
+	// One of ID, SharedImage or Marketplace should be set.
+	//
+	// If no image is specified, the image will be defaulted by the controller and Defaulted set to true which will
+	// inform the controller that if the parent MachinePool.Spec.Template.Spec.Version field is updated, this image
+	// will be updated with the corresponding default image. If Defaulted is set to false, the controller will not
+	// update the image reference when the MachinePool.Spec.Template.Spec.Version changes.
+	AzureDefaultingImage struct {
+		infrav1.Image `json:",omitempty"`
+
+		// Defaulted informs the controller that the image reference was defaulted so that it can be updated by changes
+		// to the MachinePool.Spec.Template.Spec.Version field by default.
+		// +optional
+		Defaulted bool `json:"defaulted,omitempty"`
```

Comment:
What's the UX for this? Adding this as a spec field means it can technically be set by the user. 

Is it essential to achieving the goal of this PR? If not, would you consider taking this change out and separating it to be is own PR? This would allow us to align the other uses of Image and to carefully consider the UX and back compat implications of a change like this instead of having it be a small part in this XXL PR.

Reply:
The UI is that the Defaulted can be set or unset by the user. If Default == true, then the default image version will be used for that version of K8s. If false, then a specific image reference will be needed.

This will be essential for passing the upgrade test as it is today where a machine pool's version is set and the machine pool is expected to rollout a new version of K8s.

Reply:
I've added a test to better describe the behavior of this change.

Reply:
Replaced with status tracking after design discussion.